comment_scan/message_collection.rb

```diff
@@ -1,36 +1,37 @@
 class MessageCollection
   def initialize
-    @messages = {}
+    self.clear
+  end
+
+  def clear #Clear/reset everything
+    @messages = Hash.new { |hash, key| hash[key] = [] } #Automagically create an array for each new key
     @howgoods = {}
+    @hotposts = []
+  end
+
+  def push_hot_post(post_id)
+    @hotposts.push(post_id)
+    @hotposts.shift if @hotposts.length > 10 #remove first hot_post
```

Comment:
If you want `@hotposts` to always be 10 or shorter, I'd do `@hotposts = @hotposts[0..10]`, because it makes it a bit easier to see the intent.

Reply:
Your second line is redundant here, `@hotposts.length` will never be over 10 because you've just trimmed it to have only 10 elements.

Reply:
Yikes! Must've left that in there after making that last change. Good catch!